gui/fpgaviewwidget.cc

```diff
@@ -644,7 +644,10 @@ void FPGAViewWidget::mousePressEvent(QMouseEvent *event)
     if (io.WantCaptureMouse)
         return;
 
-    if (event->buttons() & Qt::RightButton || event->buttons() & Qt::MidButton) {
+    bool shift = QApplication::keyboardModifiers().testFlag(Qt::ShiftModifier);
```

Comment:
How about refactoring this slightly to reduce the mess in the following `if()` statements? Something like:
```
bool shift = ...;
bool left = ...;
bool mid = ...;
bool right = ...;

if (left || mid || (left && shift) ...
if (left && !shift) ...
```

Also I think the second if should be `left && !shft`, otherwise you're executing a click action on shift-drag.